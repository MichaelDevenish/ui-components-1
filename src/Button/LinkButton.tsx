import React from 'react'
import { buttonClass, IButtonProps } from './ButtonHelper'
import { Anchor, AnchorProps } from '../Anchor'
import { BaseButton } from './BaseButton'

export interface LinkButtonProps extends IButtonProps {
  /** Weather the button is clickable or not */
  disabled?: boolean
  /** Destination url */
  href: string
}

export class LinkButton extends React.PureComponent<LinkButtonProps & AnchorProps> {
  public static defaultProps: Partial<LinkButtonProps & IButtonProps & AnchorProps> = {
    disabled: false,
    type: 'neutral'
  }

  public render (): JSX.Element | null {
    const {
      className,
      size,
      type,
      disabled,
      href,
      ...props
    } = this.props

    return (
      <BaseButton
        {...this.props}
        render={(content: any) => (
          <Anchor href={disabled ? '#' : href} {...props}>
            <span className={buttonClass(type!, size, className, { disabled: disabled })}>
              {content}
            </span>
          </Anchor>
        )}
      />
    )
  }
}
