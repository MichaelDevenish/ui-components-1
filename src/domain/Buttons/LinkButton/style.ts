import React from 'react'
import styled from 'styled-components'

import { Props } from '../../../common'
import { styleForMargins } from '../../Spacers/services/margins'

const StyledSpan = styled.span<{margins?: Props.IMargins}>`
   ${(props: {margins?: Props.IMargins}) => styleForMargins(props.margins)}
`

export {
  StyledSpan
}
