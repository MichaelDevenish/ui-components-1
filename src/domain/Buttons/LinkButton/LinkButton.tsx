import classNames from 'classnames'
import React, { MouseEvent } from 'react'

import { Props } from '../../../common'
import { Anchor } from '../../Internals/Anchor'
import { BaseButton, IBaseButtonProps } from '../BaseButton'
import { StyledSpan } from './style'

interface ILinkButtonProps extends IBaseButtonProps {
  /** Weather the button is clickable or not */
  disabled?: boolean,
  /** Destination url */
  href?: string,
  /** Anchor props passthrough */
  anchorComponentProps?: {
    [i: string]: any
  }
  /** The margins around the component */
  margins?: Props.IMargins
}

class LinkButton extends BaseButton<ILinkButtonProps> {
  public static defaultProps: Partial<ILinkButtonProps> = {
    ...BaseButton.defaultProps,
    disabled: false
  }

  public render (): JSX.Element | null {
    const {
      disabled,
      href,
      anchorComponentProps,
      innerRef,
      componentContext,
      margins
    } = this.props

    return (
      <Anchor
        href={href}
        anchorComponentProps={anchorComponentProps}
        onClick={this.handleClick}
      >
        <StyledSpan
          data-component-type={Props.ComponentType.LinkButton}
          data-component-context={componentContext}
          className={classNames(this.buttonClass, { disabled })}
          ref={innerRef}
          margins={margins}
        >
          {this.buttonContent}
        </StyledSpan>
      </Anchor>
    )
  }

  private handleClick = (event: MouseEvent<HTMLAnchorElement>) => {
    const {
      disabled,
      onClick
    } = this.props

    if (onClick) {
      onClick(event)
    }

    if (disabled) {
      event.preventDefault()
    }

    event.stopPropagation()
  }
}

export {
  ILinkButtonProps,
  LinkButton
}
