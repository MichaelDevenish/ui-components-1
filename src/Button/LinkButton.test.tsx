import { shallow } from 'enzyme'
import React from 'react'

import { LinkButton } from './LinkButton'

describe('<LinkButton />', () => {
  it(`should render a link button with a simple text`, () => {
    const wrapper = shallow(
      <LinkButton>
        testing testing 123
      </LinkButton>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it(`should render a disabled link button`, () => {
    const wrapper = shallow(
      <LinkButton disabled>
        testing testing 123
      </LinkButton>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it(`should render a link button with a type delete-subtle`, () => {
    const wrapper = shallow(
      <LinkButton type='delete-subtle'>
        testing testing 123
      </LinkButton>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it(`should render a link button with a size large`, () => {
    const wrapper = shallow(
      <LinkButton size='large'>
        testing testing 123
      </LinkButton>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
