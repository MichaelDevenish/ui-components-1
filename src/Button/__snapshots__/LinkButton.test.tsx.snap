// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkButton /> should render a disabled link button 1`] = `
<Anchor
  href="#"
  id="test-link-button-disabled"
>
  <span
    className="button neutral disabled"
  >
    testing testing 123
  </span>
</Anchor>
`;

exports[`<LinkButton /> should render a link button with a simple text 1`] = `
<Anchor
  href="#"
  id="test-link-button-simple"
>
  <span
    className="button neutral"
  >
    testing testing 123
  </span>
</Anchor>
`;

exports[`<LinkButton /> should render a link button with a size large 1`] = `
<Anchor
  href="#"
  id="test-link-button-large"
>
  <span
    className="button large neutral"
  >
    testing testing 123
  </span>
</Anchor>
`;

exports[`<LinkButton /> should render a link button with a type delete-subtle 1`] = `
<Anchor
  href="#"
  id="test-link-button-delete"
>
  <span
    className="button delete-subtle"
  >
    testing testing 123
  </span>
</Anchor>
`;

exports[`<LinkButton /> should render a link button with an icon 1`] = `
<Anchor
  href="test"
  icon={
    <FontAwesomeIcon
      type="star"
    />
  }
  id="test-link-button-icon"
>
  <span
    className="button primary"
  >
    <span
      className="button-icon"
      key="test-link-button-icon"
    >
      <FontAwesomeIcon
        type="star"
      />
    </span>
    Link Button with Icon
  </span>
</Anchor>
`;
